lte/gateway/c/core/oai/test/ngap/test_ngap_flows.cpp

```diff
@@ -1413,4 +1415,92 @@ TEST_F(NgapFlowTest, test_ue_notifications_from_amf) {
+
+  std::vector<MessagesIds> expected_Ids{NGAP_INITIAL_UE_MESSAGE,
+                                        NGAP_GNB_DEREGISTERED_IND};
+
+  // Verify sctp association is successful
+  EXPECT_EQ(ngap_handle_new_association(state, &peerInfo), RETURNok);
+  // Verify number of connected gNB's is 1
+  EXPECT_EQ(state->gnbs.num_elements, 1);
+
+  Ngap_NGAP_PDU_t decoded_pdu = {};
+  uint16_t length = sizeof(initial_ue_message_hexbuf) / sizeof(unsigned char);
+  bstring ngap_initial_ue_msg = blk2bstr(initial_ue_message_hexbuf, length);
+
+  // Check if the pdu can be decoded
+  ASSERT_EQ(ngap_amf_decode_pdu(&decoded_pdu, ngap_initial_ue_msg), RETURNok);
+
+  // check if initial UE message is handled successfully
+  EXPECT_EQ(ngap_amf_handle_message(state, peerInfo.assoc_id,
+                                    peerInfo.instreams, &decoded_pdu),
+            RETURNok);
+
+  gnb_description_t* gnb_association = NULL;
+  gnb_association = ngap_state_get_gnb(state, peerInfo.assoc_id);
+  ASSERT_TRUE(gnb_association != NULL);
+  EXPECT_EQ(gnb_association->ng_state, NGAP_INIT);
+  EXPECT_EQ(gnb_association->nb_ue_associated, 1);
+
+  container =
+      &(decoded_pdu.choice.initiatingMessage.value.choice.InitialUEMessage);
+  Ngap_InitialUEMessage_IEs_t* ie = NULL;
+  NGAP_TEST_PDU_FIND_PROTOCOLIE_BY_ID(Ngap_InitialUEMessage_IEs_t, ie,
+                                      container,
+                                      Ngap_ProtocolIE_ID_id_RAN_UE_NGAP_ID);
+
+  // Check if Ran_UE_NGAP_ID is present in initial message
+  ASSERT_TRUE(ie != NULL);
+  gnb_ue_ngap_id_t gnb_ue_ngap_id = 0;
+  gnb_ue_ngap_id = (gnb_ue_ngap_id_t)(ie->value.choice.RAN_UE_NGAP_ID);
+
+  // Mocking the AMF_APP_NGAP_AMF_UE_ID_NOTIFICATION from AMF
+  itti_amf_app_ngap_amf_ue_id_notification_t notification_p;
+  memset(&notification_p, 0,
+         sizeof(itti_amf_app_ngap_amf_ue_id_notification_t));
+  notification_p.gnb_ue_ngap_id = gnb_ue_ngap_id;
+  notification_p.amf_ue_ngap_id = 1;
+  notification_p.sctp_assoc_id = gnb_association->sctp_assoc_id;
+
+  ngap_handle_amf_ue_id_notification(state, &notification_p);
+
+  // Mocking the SCTP_CLOSE_ASSOCIATION ITTI message
+  sctp_message_p = itti_alloc_new_message(TASK_SCTP, SCTP_CLOSE_ASSOCIATION);
+
+  SCTP_CLOSE_ASSOCIATION(sctp_message_p).assoc_id = peerInfo.assoc_id;
+  // To test SCTP Shutdown for NGAP
+  SCTP_CLOSE_ASSOCIATION(sctp_message_p).reset = false;
+
+  EXPECT_EQ(ngap_handle_sctp_disconnection(
+                state, SCTP_CLOSE_ASSOCIATION(sctp_message_p).assoc_id,
+                SCTP_CLOSE_ASSOCIATION(sctp_message_p).reset),
+            RETURNok);
+
+  gnb_description_t* gnb_association_sd = NULL;
+  gnb_association_sd = ngap_state_get_gnb(state, peerInfo.assoc_id);
+  ASSERT_TRUE(gnb_association_sd != NULL);
+  EXPECT_EQ(gnb_association_sd->ng_state, NGAP_SHUTDOWN);
+  EXPECT_EQ(gnb_association->nb_ue_associated, 1);
+
+  EXPECT_TRUE(expected_Ids == NGAPClientServicer::getInstance().msgtype_stack);
```

Comment:
In response to SCTP disconnect indication, AMF shall release the ngap-siganling connection. So can we send "ngap_ue_context_release_command" and check that all UE contexts at NGAP are removed and gNB context is also removed.

Reply:
Addressed Review Comment